File: Proyecto Intermodular/models/product.js
'use strict'
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const ProductSchema = Schema({
	id: {
		type:String, 
	    required: true, 
	    unique: true
		},
	referencia: {
		type:String,
		required: true,
	    unique: true
		},
	nombre: {
		type:String,  
		required: true, 
		minlength:[3,"Nombre muy corto"], 
		maxlength:[15,"Nombre muy largo"]
		},
	modelo: {
		type:String,  
		required: true, 
		minlength:[15,"Nombre muy corto"], 
		maxlength:[50,"Nombre muy largo"]
		},
	descripcion: {
		type:String,  
		required: true,
		minlength:[15,"Nombre muy corto"], 
		maxlength:[250,"Nombre muy largo"]
		},
	categoria: {
		type:String, 
		required: true, 
		unique: true, 
		enum: ['phones','computers','accesories']
		},
	importe: {
		type:Number,
		required: true, 
		min:[0,"No puede ser negativo"]
		},
	iva: {
		type:Number,
		default:21
	},
	proveedor: {
		type:String,
		required: true,
	},
	tienda_stock: []
});

	ProductSchema.path('tienda_stock').validate(function(value) {
  return value.length;
},"tienda_stock' no puede estar vacio");

module.exports = mongoose.model('product', ProductSchema);

File: Proyecto Intermodular/app escritorio/WpfApplication1/WpfApplication1/Product.cs
using Newtonsoft.Json;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace App_Escritorio
{
    class Product
    {
        [JsonProperty("_id")]
        public String _id { get; set; }
        [JsonProperty("id")]
        public String id { get; set; }
        [JsonProperty("referencia")]
        public String referencia { get; set; }
        [JsonProperty("nombre")]
        public String nombre { get; set; }
        [JsonProperty("modelo")]
        public String modelo { get; set; }
        [JsonProperty("descripcion")]
        public String descripcion { get; set; }
        [JsonProperty("categoria")]
        public String categoria { get; set; }
        [JsonProperty("importe")]
        public String importe { get; set; }
        [JsonProperty("iva")]
        public String iva { get; set; }
        [JsonProperty("proveedor")]
        public String proveedor { get; set; }
        [JsonProperty("tienda_stock")]
        public ArrayList tienda_stock { get; set; }
    }
}
